### PR TITLE
8271930: Simplify end_card calculation in G1BlockOffsetTablePart::verify

### PR DESCRIPTION
**Pull request by albertnetymk**
Simplify `end_card` calculation and some general cleanup.

Test: hotspot_gc

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8271930](https://bugs.openjdk.java.net/browse/JDK-8271930): Simplify end_card calculation in G1BlockOffsetTablePart::verify

### Reviewers
 * [Thomas Schatzl](https://openjdk.java.net/census#tschatzl) (@tschatzl - **Reviewer**) ⚠️ Review applies to 82b0533c148a369ac179b948c672f78f03f6102c
 * [Ivan Walulya](https://openjdk.java.net/census#iwalulya) (@walulyai - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5015/head:pull/5015` \
`$ git checkout pull/5015`

Update a local copy of the PR: \
`$ git checkout pull/5015` \
`$ git pull https://git.openjdk.java.net/jdk pull/5015/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5015`

View PR using the GUI difftool: \
`$ git pr show -t 5015`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5015.diff">https://git.openjdk.java.net/jdk/pull/5015.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/gc/g1/heapRegion.cpp [hunk 1]
```diff
@@ -714,7 +714,8 @@ void HeapRegion::verify(VerifyOption vo,
     p += obj_size;
   }
 
-  if (!is_empty()) {
+  // only regions in old generation contain valid BOT
```
Comment (tschatzl) [inline src/hotspot/share/gc/g1/heapRegion.cpp:717]: Please start the sentence with upper case and end with full stop.

**Review by tschatzl — APPROVED** [review]
Lgtm apart from the minor comment typos.

File: src/hotspot/share/gc/g1/heapRegion.cpp (hunk 1 above)
Comment (albertnetymk) [reply src/hotspot/share/gc/g1/heapRegion.cpp]: Fixed.

**Review by walulyai — APPROVED** [review]
Lgtm!

**Comment by albertnetymk** [comment]
Thanks for the review.

/integrate

**Pull request closed without merging**